### PR TITLE
Add a listener on terminal connection

### PR DESCRIPTION
**Pull request by EtienneLt**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Feature

**What is the new behavior (if this is a feature change)?**
We are notified when a terminal is connected of disconnected.

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/NetworkListener.java [hunk 1]
```diff
@@ -36,6 +36,14 @@ default void onElementRemoved(Identifiable identifiable, String attribute, Objec
         // empty default implementation
     }
 
+    default void onElementConnected(Connectable<?> connectable) {
+        // empty default implementation
+    }
+
+    default void onElementDisconnected(Connectable<?> connectable) {
```
Comment (EtienneLt) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/NetworkListener.java:43]: I'm not sure of what is needed here ? Connectable is enough ?

Comment (geofjamg) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/NetworkListener.java:43]: Why do we not use an attribute like fir the other notifications? Why a specific method?

Comment (EtienneLt) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/NetworkListener.java]: done

**Review by geofjamg — CHANGES_REQUESTED** [review]
Could you add a unit test? (one with a bus/breaker network and one with a node/breaker network?)

**Review by geofjamg — APPROVED** [review]

**Pull request merged**